File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/Dialog.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import lombok.Builder;

@Builder
public record Dialog(
    String headerTitle,
    Component header,
    Component content,
    Component footer,
    boolean noPadding,
    boolean modeless,
    String top,
    String left,
    boolean draggable,
    String width,
    String height,
    boolean resizable,
    boolean closeButtonOnHeader,
    String style,
    String cssClasses)
    implements Component {}

File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/Dialog.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";
import Component from "@mateu/shared/apiClients/dtos/Component";
import {ComponentState} from "@infra/ui/renderers/types.ts";

export default interface Dialog extends ComponentMetadata {

    id: string
    headerTitle: string
    header: Component
    content: Component
    footer: Component
    noPadding: boolean
    modeless: boolean
    top: string
    left: string
    right: string
    draggable: boolean
    width: string
    height: string
    resizable: boolean
    closeButtonOnHeader: boolean
    initialData: ComponentState

}